polygeist/lib/Dialect/Polygeist/Transforms/BareMemRefToLLVM.cpp

```diff
@@ -473,7 +473,7 @@ void mlir::polygeist::populateBareMemRefToLLVMConversionPatterns(
   // one added by default.
   converter.addConversion([&](MemRefType type) -> Optional<Type> {
     if (!canBeLoweredToBarePtr(type))
-      return std::nullopt;
+      return Type{};
```

Comment:
This makes conversion fail for non-convertible memrefs

Reply:
Can we add `emitError`?

Reply:
Done!